### PR TITLE
Building Docker image from git source

### PR DESCRIPTION
**Pull request by AhrazA**
We ran into issues attempting to match the version of backup-utils with the version of GitHub Enterprise we were running. This is because the Dockerfile always pulls the version tagged as "stable", regardless of which version of the repo is checked out.

By building from the git source files, instead of downloading the current release tagged "stable", users will be able to easily select which version of the backup-utils they wish to build.

As it stands, the only way to specify a version would be to roll a custom Dockerfile that either builds from source, or pulls in a specific release tag from `https://github.com/github/backup-utils/archive/`.

### CODE REVIEW

**Comment by AhrazA** [comment]
Any updates regarding this?

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-*
-!share/github-backup-utils/ghe-docker-init
```
Comment (djdefi) [inline .dockerignore:2]: I think we could just delete this entire file.

**Review by djdefi — COMMENTED** [review]
Overall I think am ok with this approach, however two concerns come to mind:

1. The image size will be a bit larger, and builds will take a bit longer as Git history will get sent to the build context
2. This may change what folks using the Docker image are currently expecting to happen (i.e. run the `stable` branch).

cc: @github/backup-utils @github/ghes-infrastructure any thoughts?

**Comment by AhrazA** [comment]
Thank you for the reply @djdefi 

I made some changes that address your first comment. Also applied the same patch to the alpine Dockerfile.

Any guidance on why the macOS test might be failing? I spent some time trying to figure it out to no avail.

**Comment by AhrazA** [comment]
Pinging here again to keep this alive.

**Comment by cainejette** [comment]
cc @github/backup-utils @github/ghes-infrastructure

**Comment by djdefi** [comment]
> Any guidance on why the macOS test might be failing? I spent some time trying to figure it out to no avail.

I think this may be due to MacOS's default version of `stat` not supporting the same flags as the GNU Linux `stat`.

**Review by zachary-mark — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**